ext/mbstring/tests/bug43994.phpt

```diff
@@ -49,7 +49,8 @@ With $regs arg:
 
 Warning: mb_ereg(): empty pattern in %s on line %d
 bool(false)
-NULL
+array(0) {
+}
```

Comment:
@cmb69 The function does not yield null, it just doesn't modify the `zval` you passed as an argument when the match fails. If it was a string it stays a string, if it was not defined it will be null, etc...

This behavior is not consitent with `preg_match`, is highly counter-intuitive and looks like a bug to me:

```php
$pattern = '(a)\1';
mb_ereg($pattern, 'aa', $matches);
// $matches = ['aa', 'a']
mb_ereg($pattern, 'WTF?', $matches);
// $matches = ['aa', 'a']

preg_match($pattern, 'aa', $matches);
// $matches = ['aa', 'a']
preg_match($pattern, 'WTF?', $matches);
// $matches = []
```